src/Institution/Institution.php

```diff
@@ -1,39 +1,127 @@
+    const streetAddress = 'streetAddress';
+    const locality = 'locality';
+    const postalCode = 'postalCode';
+    const countryName = 'countryName';
+    const enableStatusesSystem = 'enableStatusesSystem';
+    const enableSkosXl = 'enableSkosXl';
 
     /**
-     * @var Triple[]
+     * @var string[]
      */
-    private $triples = [];
+    private static $mapping = [
+        self::code => OpenSkos::CODE,
+        self::name => OpenSkos::NAME,
+        self::organisationUnit => VCard::ORGUNIT,
+        self::email => VCard::EMAIL,
+        self::website => OpenSkos::WEBPAGE,
+        self::streetAddress => VCard::ADR,
+        self::locality => VCard::LOCALITY,
+        self::postalCode => VCard::PCODE,
+        self::countryName => VCard::COUNTRY,
+        self::enableStatusesSystem => OpenSkos::ENABLESTATUSSESSYSTEMS,
+        self::enableSkosXl => OpenSkos::ENABLESKOSXL,
+    ];
 
-    private function __construct() {}
+    /**
+     * @var Iri
+     */
+    private $subject;
 
     /**
-     * @return string[]
+     * @var Literal[]
      */
-    public function properties() : array
+    private $literals = [];
+
+    public function __construct(Iri $subject)
     {
-        $res = [];
-        foreach ($this->triples as $triple) {
-            $res[$triple->getPredicate()] = $triple->getObject();
-        }
+        $this->subject = $subject;
+    }
+
+    //TODO: Generate getters.
 
-        return $res;
+    /**
+     * @return Iri
+     */
+    public function getSubject(): Iri
+    {
+        return $this->subject;
     }
 
     /**
+     * @return Literal|null
+     */
+    public function getCode(): ?Literal
+    {
+        return $this->literals[self::code] ?? null;
+    }
+
+    /**
+     * @return Literal|null
+     */
+    public function getWebsite(): ?Literal
+    {
+        return $this->literals[self::website] ?? null;
+    }
+
+    /**
+     * @param Iri      $subject
      * @param Triple[] $triples
+     *
+     * @return Institution
      */
-    public static function fromTriples(array $triples) : self
+    public static function fromTriples(Iri $subject, array $triples): Institution
     {
-        $obj = new self();
-        $obj->triples = $triples;
+        $invMapping = array_flip(self::$mapping);
+        $literals = [];
+        foreach ($triples as $triple) {
+            // Skip unrelated triples
+            if ($triple->getSubject()->getUri() !== $subject->getUri()) {
+                continue;
+            }
+
+            $property = $invMapping[$triple->getPredicate()->getUri()] ?? null;
+            // Skip unknown properties
+            if (null === $property) {
+                continue;
+            }
+
+            $object = $triple->getObject();
+            // Skip non-literals. TODO: throw an exception?
+            if (!$object instanceof Literal) {
+                continue;
+            }
+            $literals[$property] = $object;
+
+            /* TODO: Add Resource when needed
```

Comment:
Probably better to not add TODO's in the code we should either implement it or not.

Reply:
Nope, that is a good thing to keep all technical dept as FIXME/TODO in the code. 
Then it is easier to navigate and track. And also use nice tools such as https://imdone.io/
That particular todo:  `Add Resource when needed` seems useless because of that when needed.  So it can be removed. but all good suggestions I would like to have them registered either as TODO (tactical improvements) or as an issue (strategic improvements).